### PR TITLE
fix(designs): render imported designs misclassified as legacy v1alpha2

### PR DESCRIPTION
**Pull request by leecalcote**
Imported designs (Kubernetes manifests, Helm charts, Compose files) are produced by NewPatternFileFromK8sManifest in v1beta3 form. GetMesheryPatternHandler gated the legacy v1alpha2->v1beta3 migration on IsDesignInAlpha2Format, which recognized only v1beta1 as a current schema. Every freshly imported v1beta3 design therefore failed the check, was treated as alpha2, and convertV1alpha2ToV1beta3 reparsed it as a v1alpha2.PatternFile (which carries 'services', not 'components') - yielding a design with zero components that rendered as an empty canvas in Kanvas. The POST get-handler path additionally re-persisted the emptied design, corrupting it in storage.

Detect alpha2 by the absence of a canonical 'designs.meshery.io/*' schemaVersion (legacy alpha2 carries a top-level 'services' map and no versioned schemaVersion) instead of whitelisting a single current version, so current and future design schema versions are never routed through the lossy migration. Add a regression test covering v1beta3/v1beta1/future-version (current) and v1alpha2 (legacy).

Signed-off-by: Lee Calcote <lee.calcote@layer5.io>

### CODE REVIEW

**Comment by leecalcote** [comment]
@fitzergerald will you have a look?

**Review by fitzergerald — DISMISSED** [review]
lgtm

File: server/models/pattern/utils/utils.go [hunk 1]
```diff
@@ -112,15 +111,30 @@ func GetRandomAlphabetsOfDigit(length int) (s string) {
...
-	if err != nil {
+	if err := encoding.Unmarshal([]byte(patternFile), &design); err != nil {
 		return true, err
 	}
 
-	val, ok := design["schemaVersion"]
-	if ok && val == v1beta1.DesignSchemaVersion {
+	if sv, ok := design["schemaVersion"].(string); ok && strings.HasPrefix(sv, "designs.meshery.io/") {
 		return false, nil
 	}
```
[the remark this answers is not in the document]
Comment (leecalcote) [reply server/models/pattern/utils/utils.go]: Adopted - now decoding only `schemaVersion` into a targeted struct instead of `map[string]interface{}`, avoiding the full-tree allocation on every design read.

One correction to the snippet: the struct needs a `yaml` tag as well as `json`. `encoding.Unmarshal` tries `encoding/json` first and falls back to `gopkg.in/yaml.v3`, which keys off `yaml` tags (or the lowercased field name), not `json` tags. So a `json`-only tag misses the camelCase `schemaVersion` key in a YAML-encoded pattern file, leaving it empty and misclassifying the design as alpha2 - the same regression this PR fixes, reintroduced for YAML.

Verified: `json`-only returns "" for a YAML pattern file; `json`+`yaml` returns the value. Pushed in 6bde120 with both tags plus a YAML regression test.

**Comment by leecalcote** [comment]
Feedback addressed across `6bde120` and `6e8723e`:

**Performance (inline review):** Adopted - `IsDesignInAlpha2Format` now decodes only `schemaVersion` into a targeted struct instead of `map[string]interface{}`. One correction to the suggested snippet: the struct needs **both** `json` and `yaml` tags. `encoding.Unmarshal` tries `encoding/json` first and falls back to `gopkg.in/yaml.v3`, which keys off `yaml` tags (not `json`), so a `json`-only tag returns `""` for a YAML-encoded pattern file and would reintroduce the misclassification. Added a YAML regression test alongside the JSON cases.

**Edge-case tests:** Added `missing schemaVersion key` and `empty schemaVersion` rows - both classify as legacy (migrate), as predicted.

**Stale comment:** Replaced the `// v1beta1` comment above the `IsDesignInAlpha2Format` call in `meshery_pattern_handler.go` with one describing the legacy-v1alpha2 migration. The other two call sites (lines 1310 / 1609) carried no stale comment.

**E2E run (1 failed):** Unrelated to this change. The failure is `Create a Model` (10-minute timeout - infra/flake), plus three RBAC relationship cases (`ClusterRole`/`Role` → `ServiceAccount` permission bindings and a `ClusterRoleBinding` → `ClusterRole` reference). This change only affects design schema-version classification on read - orthogonal to model creation and relationship evaluation. The same run confirms design loading is healthy here: alias resolution, config patching, evaluation idempotency, structural integrity, and the `meshery-design` / `service-to-deployment-network` relationship identifications all pass. The three RBAC failures are pre-existing.

**Review by yi-nuo426 — APPROVED** [review]
